contrib/pg_tde/documentation/docs/functions.md

```diff
@@ -1,97 +1,254 @@
+
+* `pg_tde_add_key_provider_<type>('provider-name', <provider specific parameters>)`
+* `pg_tde_add_global_key_provider_<type>('provider-name', <provider specific parameters>)`
+
+When you add a new provider, the provider name must be unqiue in the scope. But a local database provider and a global provider can have the same name.
+
+### Change an existing provider
 
-## pg_tde_add_key_provider_vault_v2
+You can change an existing key provider using the provided functions, which are implemented for each provider type.
 
-Creates a new key provider for the database using a remote HashiCorp Vault server.
+There are two functions to change existing providers: one to change a provider in the current database, and another one to change a provider in the global scope.
 
-The specified access parameters require permission to read and write keys at the location.
+* `pg_tde_change_key_provider_<type>('provider-name', <provider specific parameters>)`
+* `pg_tde_change_global_key_provider_<type>('provider-name', <provider specific parameters>)`
+
+When you change a provider, the referred name must exist in the database local or a global scope.
+
+The `change` functions require the same parameters as the `add` functions. They overwrite the setting for every parameter except for the name, which can't be changed.
+
+Provider specific parameters differ for each implementation. Refer to the  respective subsection for details.
+
+**Some provider specific parameters contain sensitive information, such as passwords. Never specify these directly, use the remote configuration option instead.**
+
+#### Adding or modifying Vault providers
+
+The Vault provider connects to a HashiCorp Vault or an OpenBao server, and stores the keys on a key-value store version 2.
+
+Use the following functions to add the Vault provider:
 
 ```
 SELECT pg_tde_add_key_provider_vault_v2('provider-name','secret_token','url','mount','ca_path');
+SELECT pg_tde_add_global_key_provider_vault_v2('provider-name','secret_token','url','mount','ca_path');
+```
+
+These functions change the Vault provider:
+
+```
+SELECT pg_tde_change_key_provider_vault_v2('provider-name','secret_token','url','mount','ca_path');
+SELECT pg_tde_change_global_key_provider_vault_v2('provider-name','secret_token','url','mount','ca_path');
 ```
 
 where:
 
+* `provider-name` is the name of the key provider
 * `url` is the URL of the Vault server
-* `mount` is the mount point where the keyring should store the keys
+* `mount` is the mount point on the Vault server where the key provider should store the keys
 * `secret_token` is an access token with read and write access to the above mount point
 * [optional] `ca_path` is the path of the CA file used for SSL verification
 
 All parameters can be either strings, or JSON objects [referencing remote parameters](external-parameters.md).
 
-## pg_tde_add_key_provider_kmip
+**Never specify the secret token directly, use a remote parameter instead.**
```

Comment:
If that is the case why do we even let users do so? Just food for thoughts, nothing we need to fix.